apollo-client/src/main/java/com/ctrip/framework/apollo/spring/annotation/ApolloAnnotationProcessor.java

```diff
@@ -54,21 +56,35 @@ protected void processMethod(final Object bean, String beanName, final Method me
     ReflectionUtils.makeAccessible(method);
     String[] namespaces = annotation.value();
     String[] annotatedInterestedKeys = annotation.interestedKeys();
-    Set<String> interestedKeys = annotatedInterestedKeys.length > 0 ? Sets.newHashSet(annotatedInterestedKeys) : null;
+    String[] annotatedInterestedKeyPrefixes = annotation.interestedPrefixes();
     ConfigChangeListener configChangeListener = new ConfigChangeListener() {
       @Override
       public void onChange(ConfigChangeEvent changeEvent) {
         ReflectionUtils.invokeMethod(method, bean, changeEvent);
       }
     };
 
+    Set<String> interestedKeys = null;
+    Set<String> interestedKeyPrefixes = null;
+    if (annotatedInterestedKeys.length > 0) {
+      interestedKeys = new HashSet<>(Arrays.asList(annotatedInterestedKeys));
+    }
+    if (annotatedInterestedKeyPrefixes.length > 0) {
+      interestedKeyPrefixes = new HashSet<>(Arrays.asList(annotatedInterestedKeyPrefixes));
+    }
+
     for (String namespace : namespaces) {
       Config config = ConfigService.getConfig(namespace);
 
-      if (interestedKeys == null) {
+      if (interestedKeys == null && interestedKeyPrefixes == null) {
         config.addChangeListener(configChangeListener);
       } else {
-        config.addChangeListener(configChangeListener, interestedKeys);
+        if (interestedKeys != null && !interestedKeys.isEmpty()) {
```

Comment:
how about addChangeListener once instead of twice?

```java
config.addChangeListener(configChangeListener, interestedKeys, interestedKeyPrefixes);
```